From 96594ccd8805f150ea29596db8b5993f0e491f78 Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Sat, 21 Jan 2017 20:22:44 +0000
Subject: [PATCH] [CONFIGURATION-649] Documentation updates.

Added a warning about changing the list delimiter handler to both
Javadocs and the user's guide.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/trunk@1779754 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/configuration2/AbstractConfiguration.java  | 10 ++++++++++
 src/site/xdoc/userguide/howto_basicfeatures.xml        |  8 ++++++++
 2 files changed, 18 insertions(+)

diff --git a/src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java b/src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java
index 4adbd75de6..09966be758 100644
--- a/src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java
+++ b/src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java
@@ -149,6 +149,7 @@ public ListDelimiterHandler getListDelimiterHandler()
     }
 
     /**
+     * <p>
      * Sets the {@code ListDelimiterHandler} to be used by this instance. This
      * object is invoked every time when dealing with string properties that may
      * contain a list delimiter and thus have to be split to multiple values.
@@ -156,6 +157,15 @@ public ListDelimiterHandler getListDelimiterHandler()
      * does not support list splitting. This can be changed for instance by
      * setting a {@link org.apache.commons.configuration2.convert.DefaultListDelimiterHandler
      * DefaultListDelimiterHandler} object.
+     * </p>
+     * <p>
+     * <strong>Warning:</strong> Be careful when changing the list delimiter
+     * handler when the configuration has already been loaded/populated. List
+     * handling is typically applied already when properties are added to the
+     * configuration. If later another handler is set which processes lists
+     * differently, results may be unexpected; some operations may even cause
+     * exceptions.
+     * </p>
      *
      * @param listDelimiterHandler the {@code ListDelimiterHandler} to be used
      *        (must not be <b>null</b>)
diff --git a/src/site/xdoc/userguide/howto_basicfeatures.xml b/src/site/xdoc/userguide/howto_basicfeatures.xml
index 821f5c9bbe..8e7baf2513 100644
--- a/src/site/xdoc/userguide/howto_basicfeatures.xml
+++ b/src/site/xdoc/userguide/howto_basicfeatures.xml
@@ -173,6 +173,14 @@ String firstPieColor = config.getString("colors.pie");
       <code>getString()</code> method is called for a property that has
       multiple values. This call will return the first value of the list.
     </p>
+    <p>
+      One word of warning at the end: Be careful when changing the list
+      delimiter handler on a configuration which already contains values. List
+      handling is typically applied already when properties are added to the
+      configuration. If later another handler is set which processes lists
+      differently, results may be unexpected; some operations may even cause
+      exceptions!
+    </p>
     </subsection>
 
     <subsection name="Variable Interpolation">
